src/server/game/Server/WorldSession.cpp

```diff
@@ -396,6 +396,12 @@ bool WorldSession::Update(uint32 diff, PacketFilter& updater)
 
     _recvQueue.readd(requeuePackets.begin(), requeuePackets.end());
 
+    if (_player && _player->IsRepopPending() && !_player->HasPendingMovementChange())
+        _player->RepopAtGraveyard();
+
+    if (_player && _player->IsHasDelayedTeleport() && _player->IsAlive())
```

Comment:
Wouldn't `Player::ProcessDelayedOperations` be a better place for this? (it was kind of designed for delayed tasks)

Reply:
It's not exactly what I need. I need to delay the teleport itself until some condition (in this case that all pending movement ACKs are received) is verified. In other words, I needed to execute some code after the packet has been handled and as soon as the specified condition is verified.

Reply:
That further emphasizes my point though. You are now relying on function call order "somewhere else in the code" which makes this code very likely to break if another developer decides to refactor anything.
Make the condition explicit, check that ACK queue is empty
Another thing - there is `_player` on every single part of this code chunk clearly indicating that it belongs to Player class

I think you really should move it into `Player::ProcessDelayedOperations` while adding all remaining conditions you might need

Reply:
The issue is that `Player::ProcessDelayedOperations` is not adapted to my use case because it is not called at the right moments. In the 3.3.5 branch (i don't know if it is different in master), it is only called at the end of the handler for opcodes `MSG_MOVE_WORLDPORT_ACK` and `MSG_MOVE_TELEPORT_ACK`. 

Like I said earlier, I need to execute some code after the execution of the handler has ended. Did I miss something in your explaination?

Reply:
I agree with you on the general idea though: that this code is not great, maintainability wise. I'll see what I can do.

Reply:
Oh, i thought it was called periodically like `Update`, my bad (there is no difference in master)